res/controllers/Reloop-Jockey-3-ME-scripts.js

```diff
@@ -0,0 +1,444 @@
+
+Jockey3ME.CUP = function (channel, control, value, status, group) {
+  if (value == 0x7F) {
+    engine.setValue(group,"cue_default",1);
+    engine.setValue(group,"cue_default",0);
+    midi.sendShortMsg(status,control,127);
+  } else {
+    engine.setValue(group,"play",1);
+    midi.sendShortMsg(status,control,0);
+  }
+}
+
+Jockey3ME.MixerVol = function (channel, control, value, status, group) {
+  var currentDeck = parseInt(group.substring(8,9));
+  if ((Jockey3ME.MixerDeck1 == 1 && currentDeck == 1) || (Jockey3ME.MixerDeck2 == 1 && currentDeck == 2)) {
+    currentDeck += 2;
+  }
+  if (control == 0x2D || control == 0x6C) {
+	var noGainHop = engine.getParameter("[Channel" + currentDeck + "]","pregain");
+	if (!(!((noGainHop - (value / 127)) < 0.04) || !((noGainHop - (value / 127)) > -0.04))) {
+		Jockey3ME.noVolHopValue[1] = false;
+	}
+	if (!Jockey3ME.noVolHopValue[1]) {
+		engine.setParameter("[Channel" + currentDeck + "]","pregain",(value / 127));
+    }
+  } else {
+    var noVolHop = engine.getParameter("[Channel" + currentDeck + "]","volume");
+    if (!(!((noVolHop - (value / 127)) < 0.04) || !((noVolHop - (value / 127)) > -0.04))) {
+      Jockey3ME.noVolHopValue[0] = false;
+    }
+    if (!Jockey3ME.noVolHopValue[0]) {
+      engine.setParameter("[Channel" + currentDeck + "]","volume",(value / 127));
+    }
+  }
+}
+
+Jockey3ME.DeckSwitch = function (channel, control, value, status, group) {
+  if (control == 0x3C && value == 0x7F) {
+    Jockey3ME.MixerDeck1 = 1;
+  } else if (control == 0x3C && value == 0x00) {
+    Jockey3ME.MixerDeck1 = 0;
+  } else if (control == 0x3F && value == 0x7F) {
+    Jockey3ME.MixerDeck2 = 1;
+  } else if (control == 0x3F && value == 0x00) {
+    Jockey3ME.MixerDeck2 = 0;
+  }
+  for (var i = 0; i < Jockey3ME.noVolHopValue.length; i++) {
+  	Jockey3ME.noVolHopValue[i] = true;
+  }
+  // Jockey3ME.noVolHopValue[0] = true;
+}
+
+Jockey3ME.EQ = function (channel, control, value, status, group) {
+  var currentDeck = parseInt(group.substring(24,25));
+  if ((Jockey3ME.MixerDeck1 == 1 && currentDeck == 1) || (Jockey3ME.MixerDeck2 == 1 && currentDeck == 2)) {
+    currentDeck += 2;
+  }
+  switch (control) {
+    case 0x2E:
+      var eqKnop = 3;
+      break;
+    case 0x2F:
+      var eqKnop = 2;
+      break;
+    case 0x30:
+      var eqKnop = 1;
+      break;
+    case 0x6D:
+      var eqKnop = 3;
+      break;
+    case 0x6E:
+      var eqKnop = 2;
+      break;
+    case 0x6F:
+      var eqKnop = 1;
+      break;
+    default:
+      print("Error on EQ chosing");
+  }
+  var noHighHop = engine.getParameter("[EqualizerRack1_[Channel" + currentDeck + "]_Effect1]","parameter" + eqKnop);
+  if (!(!((noHighHop - (value / 127)) < 0.04) || !((noHighHop - (value / 127)) > -0.04))) {
+  	Jockey3ME.noVolHopValue[eqKnop + 1] = false;
+  }
+  if (!Jockey3ME.noVolHopValue[eqKnop + 1]) {
+  	engine.setParameter("[EqualizerRack1_[Channel" + currentDeck + "]_Effect1]","parameter" + eqKnop, (value / 127));
+  }
+}
```

Comment:
Like the volume faders, couldn't this be done with XML with the soft-takeover option?